src/overlays/actors/ovl_Obj_HsStump/z_obj_hsstump.c

```diff
@@ -23,18 +28,132 @@ const ActorInit Obj_HsStump_InitVars = {
 };
 */
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_HsStump_0x80BDAA30/ObjHsStump_Init.asm")
+extern Gfx D_060003B8[];
+
+extern CollisionHeader D_060011B0;
+
+extern InitChainEntry D_80BDAF80[];
+/*
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 180, ICHAIN_STOP),
+};
+*/
+
+extern Vec3f D_80BDAF84;
+// static Vec3f iceSmokeAccel = {0.0f, 0.0f, 0.0f};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_HsStump_0x80BDAA30/func_80BDAB00.asm")
+void ObjHsStump_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHsStump* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_HsStump_0x80BDAA30/func_80BDAB18.asm")
+    Actor_ProcessInitChain(&this->dyna.actor, D_80BDAF80);
+    this->isHidden = OBJHSSTUMP_GET_ISHIDDEN(thisx);
+    this->switchFlag = OBJHSSTUMP_GET_SWITCHFLAG(thisx); // Must be thisx to match
+    BcCheck3_BgActorInit(&this->dyna, 1);
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_060011B0);
+    if (this->isHidden != 0) {
+        if (this->isHidden != 1) {
+            return;
+        }
+        if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+            this->isHidden = 0;
+        } else {
+            this->dyna.actor.draw = NULL;
+            Actor_SetScale(&this->dyna.actor, 0.0f);
+            func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        }
+    }
+    ObjHsStump_SetupIdle(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_HsStump_0x80BDAA30/func_80BDAB6C.asm")
+void ObjHsStump_SetupIdle(ObjHsStump* this, GlobalContext* globalCtx) {
+    this->actionFunc = ObjHsStump_Idle;
+}
 
+void ObjHsStump_Idle(ObjHsStump* this, GlobalContext* globalCtx) {
+    if (this->isHidden == 1) {
+        if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
```

Comment:
`if ((this->isHidden == 1) && Flags_GetSwitch(globalCtx, this->switchFlag) {`

Reply:
👍